### DISCUSSION TITLE
How to multiple column IN query placeholder?

### DISCUSSION THREAD

**Discussion by a-bechan** [prisma/prisma#24958] [Q&A] [answered]
### Question

I'm trying to execute the following query using `$queryRaw`.

```sql
SELECT *
FROM my_table
WHERE
  (column_1, column_2) IN (
    ('val_1', 'val_2'),
    ('val_3', 'val_4')
  )
```

However, I don't know how to specify the variable in the IN clause.

The variables I'm assuming are arrays like this:
```sql
const params = [
  ['val1', 'val2'],
  ['val3', 'val4'],
];
```

**Comment by ludralph** [comment, marked as answer]
Hi @okasyo 👋 

Can you try the code below?

```js
const params = [
    ['val1', 'val2'],
    ['val3', 'val4'],
  ];
  
const formattedParams = params.map(tuple => `(${tuple.map(val => `'${val}'`).join(', ')})`).join(', ');
  
const result = await prisma.$queryRaw`SELECT * FROM "MyTable" WHERE (column_1, column_2) IN (${Prisma.sql([formattedParams])})`;
```
`formattedParams` is a string that formats each tuple into the required SQL syntax. `Prisma.sql` is used to safely insert the formatted string into the query.

**Reply by a-bechan** [reply]
@ludralph 
It works! Thank you!
